src/Microsoft.Identity.Client/ApiEvent.cs

```diff
@@ -40,17 +41,42 @@ internal class ApiEvent : EventBase
         public const string ConstTenantId = EventNamePrefix + "tenant_id";
         public const string ConstUserId = EventNamePrefix + "user_id";
         public const string ConstWasSuccessful = EventNamePrefix + "was_successful";
+        public const string ConstCorrelationId = EventNamePrefix + "correlation_id";
+        public const string ConstRequestId = EventNamePrefix + "request_id";
+        public const string ConstIsConfidentialClient = EventNamePrefix + "is_confidential_client";
+
+        public enum ApiIds
+        {
+            AcquireTokenSilentWithAuthority = 31,
+            AcquireTokenSilentWithoutAuthority = 30,
+
+            AcquireTokenWithScope = 170,
+            AcquireTokenWithScopeHint = 171,
+            AcquireTokenWithScopeHintBehavior = 172,
+            AcquireTokenWithScopeHintBehaviorAuthority = 173,
+            AcquireTokenWithScopeUser = 176,
+            AcquireTokenWithScopeUserBehavior = 174,
+            AcquireTokenWithScopeUserBehaviorAuthority = 175,
+
+            AcquireTokenOnBehalfOfWithScopeUser = 520,
+            AcquireTokenOnBehalfOfWithScopeUserAuthority = 521,
+
+            AcquireTokenForClientWithScope = 726,
+            AcquireTokenForClientWithScopeRefresh = 727,
+
+            AcquireTokenByAuthorizationCodeWithCodeScope = 830,
+        }
 
         public ApiEvent() : base(EventNamePrefix + "api_event") {}
 
-        public int ApiId
+        public ApiIds ApiId
         {
-            set => this[ConstApiId] = value.ToString();
+            set => this[ConstApiId] = ((int)value).ToString();
         }
 
-        public string Authority
+        public Uri Authority
```

Comment:
The eventual output to the event receiver is still using string. Here you are looking at a property setter, and this setter happens to accept a Uri as input (because its underlying helper `ScrubTenant()` expects a Uri). This setter thing is considered as an implementation detail anyway. I/we are not going to change the output format defined in the internal design.